modules/plugin/jdbc/jdbc-oracle/src/main/java/org/geotools/data/oracle/OracleFilterToSQL.java

```diff
@@ -173,6 +174,9 @@ protected FilterCapabilities createFilterCapabilities() {
         caps.addType(EndedBy.class);
         caps.addType(TEquals.class);
 
+        // native filter support
+        caps.addType(NativeFilter.class);
```

Comment:
If all the native filter handling in SQL does is to take it and write it out in the SQL, why not add support for it in SQLDialect#BASE_DBMS_CAPABILITIES instead? And selected DBMS can remove it, if needs be (don't think there will be such need)

Reply:
Good point, done.